<a href="https://colab.research.google.com/github/khatriprajwol/Data-Visualization-/blob/main/Integersandfloats.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##Integers and Floats

In [56]:
import numpy as np #a matrix like package for handling data
import pandas as pd #a R like package for handling data
from scipy import stats #a way to just get a piece of a large package
import matplotlib.pyplot as plt
import seaborn as sns

In [57]:
import pandas as pa

df = pa.read_csv('https://raw.githubusercontent.com/nurfnick/Data_Viz/main/iris.csv')

df.head().T

0            1            2            3            4
SepalLength          5.1          4.9          4.7          4.6          5.0
SepalWidth           3.5          3.0          3.2          3.1          3.6
PedalLength          1.4          1.4          1.3          1.5          1.4
PedalWidth           0.2          0.2          0.2          0.2          0.2
Class        Iris-setosa  Iris-setosa  Iris-setosa  Iris-setosa  Iris-setosa

In [58]:
df.describe()

SepalLength  SepalWidth  PedalLength  PedalWidth
count   150.000000  150.000000   150.000000  150.000000
mean      5.843333    3.054000     3.758667    1.198667
std       0.828066    0.433594     1.764420    0.763161
min       4.300000    2.000000     1.000000    0.100000
25%       5.100000    2.800000     1.600000    0.300000
50%       5.800000    3.000000     4.350000    1.300000
75%       6.400000    3.300000     5.100000    1.800000
max       7.900000    4.400000     6.900000    2.500000

In [59]:
df.SepalLength.astype('int')
df.SepalLength = df.SepalLength.astype('int')
df.head().T

0            1            2            3            4
SepalLength            5            4            4            4            5
SepalWidth           3.5          3.0          3.2          3.1          3.6
PedalLength          1.4          1.4          1.3          1.5          1.4
PedalWidth           0.2          0.2          0.2          0.2          0.2
Class        Iris-setosa  Iris-setosa  Iris-setosa  Iris-setosa  Iris-setosa

In [60]:
df.SepalLength.agg(['mean','median','count', 'std'])

mean        5.386667
median      5.000000
count     150.000000
std         0.841752
Name: SepalLength, dtype: float64

Yes converting SepalLength to integer decreased the mean from 5.843333 to 5.386667. 

In [61]:
df.groupbyclass= df.groupby(by='Class')
df.head().T

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: UserWarning: Pandas doesn't allow columns to be created via a new attribute name - see https://pandas.pydata.org/pandas-docs/stable/indexing.html#attribute-access
  """Entry point for launching an IPython kernel.


0            1            2            3            4
SepalLength            5            4            4            4            5
SepalWidth           3.5          3.0          3.2          3.1          3.6
PedalLength          1.4          1.4          1.3          1.5          1.4
PedalWidth           0.2          0.2          0.2          0.2          0.2
Class        Iris-setosa  Iris-setosa  Iris-setosa  Iris-setosa  Iris-setosa

I do not see any difference even if I group by `Class`. 

In [62]:
df.groupby(by = 'Class')

In [63]:
df.groupby('Class').SepalLength.agg(['mean','median','count', 'std'])



mean  median  count       std
Class                                         
Iris-setosa      4.60     5.0     50  0.494872
Iris-versicolor  5.48     5.0     50  0.579937
Iris-virginica   6.08     6.0     50  0.665168